packages/core/src/components/entity-title/entityTitle.tsx

```diff
@@ -0,0 +1,148 @@
+/* !
+ * (c) Copyright 2023 Palantir Technologies Inc. All rights reserved.
+ */
+
+import classNames from "classnames";
+import * as React from "react";
+
+import { IconName, IconNames } from "@blueprintjs/icons";
+
+import { Classes, DISPLAYNAME_PREFIX, MaybeElement, Props } from "../../common";
+import { H1, H2, H3, H4, H5, H6 } from "../html/html";
+import { Icon } from "../icon/icon";
+import { Text } from "../text/text";
+
+export enum HeadingSize {
+    h6 = "h6",
+    h5 = "h5",
+    h4 = "h4",
+    h3 = "h3",
+    h2 = "h2",
+    h1 = "h1",
+}
+
+export type IEntityTitleProps = Props & {
+    title: JSX.Element | string;
+
+    /**
+     * Inherit font size if not provided
+     */
+    headingSize?: HeadingSize;
+
+    titleURL?: string;
+
+    /**
+     * Name of a Blueprint UI icon (or an icon element) to render in the section's header.
+     * Note that the header will only be rendered if `title` is provided.
+     */
+    icon?: IconName | MaybeElement;
+
+    /**
+     * <Tag> components work best - if multiple, wrap in <React.Fragment>
+     */
+    tags?: React.ReactNode;
+
+    subtitle?: JSX.Element | string;
+};
+
+export const EntityTitle: React.FC<IEntityTitleProps> = ({
+    className,
+    headingSize,
+    title,
+    icon,
+    tags,
+    subtitle,
+    titleURL,
+}) => {
+    const titleComponent = React.useMemo(() => {
+        const maybeTitleWithURL =
+            titleURL != null ? (
+                <a target="_blank" href={titleURL} rel="noreferrer">
+                    {title}
+                </a>
+            ) : (
+                title
+            );
+
+        const titleClassName = classNames(Classes.ENTITY_TITLE_TITLE);
+
+        switch (headingSize) {
+            case HeadingSize.h1:
+                return <H1 className={titleClassName}>{maybeTitleWithURL}</H1>;
+            case HeadingSize.h2:
+                return <H2 className={titleClassName}>{maybeTitleWithURL}</H2>;
+            case HeadingSize.h3:
+                return <H3 className={titleClassName}>{maybeTitleWithURL}</H3>;
+            case HeadingSize.h4:
+                return <H4 className={titleClassName}>{maybeTitleWithURL}</H4>;
+            case HeadingSize.h5:
+                return <H5 className={titleClassName}>{maybeTitleWithURL}</H5>;
+            case HeadingSize.h6:
+                return <H6 className={titleClassName}>{maybeTitleWithURL}</H6>;
+            default:
+                return <Text className={titleClassName}>{maybeTitleWithURL}</Text>;
+        }
+    }, [title, titleURL, headingSize]);
+
+    const subtitleComponent = React.useMemo(() => {
```

Comment:
again, this is an _element_ rather than a component value. actually you can drop the suffix altogether. it's more important to use the "maybe" prefix in this case:

```suggestion
    const maybeSubtitle = React.useMemo(() => {
```